src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -46,26 +46,21 @@ protected RemoteHostClient(Workspace workspace)
 
         protected abstract void OnStarted();
 
-        protected abstract void OnStopped();
-
-        internal void Shutdown()
-        {
-            // this should be only used by RemoteHostService to shutdown this remote host
-            Stopped();
-        }
-
         protected void Started()
         {
             OnStarted();
 
-            OnStatusChanged(true);
+            OnStatusChanged(started: true);
         }
 
-        protected void Stopped()
+        public void Dispose()
         {
-            OnStopped();
+            Dispose(disposing: true);
+        }
 
-            OnStatusChanged(false);
+        protected virtual void Dispose(bool disposing)
```

Comment:
❕  This is the legacy dispose pattern. New `IDisposable` objects should make `Dispose()` virtual and _not_ introduce `Dispose(bool)`.

Reply:
I also don't like it but I wasn't aware it's considered legacy now. There is nothing in the docs that says so: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Reply:
@sharwell Ping. Does the documentation need to be updated?

Reply:
Eventually, the documentation should be updated. I don't expect it to occur soon though. For now we should just use the new pattern where it makes sense.